x/auth/types/account.go

```diff
@@ -64,20 +72,32 @@ func (acc *BaseAccount) SetAddress(addr sdk.AccAddress) error {
 
 // GetPubKey - Implements sdk.AccountI.
 func (acc BaseAccount) GetPubKey() (pk crypto.PubKey) {
-	if len(acc.PubKey) == 0 {
+	if acc.PubKey == nil {
 		return nil
 	}
-
-	amino.MustUnmarshalBinaryBare(acc.PubKey, &pk)
-	return pk
+	content, ok := acc.PubKey.GetCachedValue().(crypto.PubKey)
+	if !ok {
+		return nil
```

Comment:
why do we return `nil` here? Shouldn't we deserialize and set cache?

Reply:
it's a getter, how would you set the cached value? not sure to get what you mean.